### PR TITLE
Replaced DeveloperError with Check in sampleTerrain.js

### PR DESCRIPTION
**Pull request by rms13**
Worked on issue #4794, replaced DeveloperError with Check in Source/Core/sampleTerrain.js

### CODE REVIEW

File: Source/Core/sampleTerrain.js [hunk 1]
```diff
@@ -45,15 +45,9 @@ define([
...
-        if (!defined(terrainProvider)) {
-            throw new DeveloperError('terrainProvider is required.');
-        }
-        if (!defined(level)) {
-            throw new DeveloperError('level is required.');
-        }
-        if (!defined(positions)) {
-            throw new DeveloperError('positions is required.');
-        }
+        Check.typeOf.object('terrainProvider',terrainProvider);
```
Comment (omh1280) [inline Source/Core/sampleTerrain.js]: Please put a space between the arguments. Thanks!

**Comment by omh1280** [comment]
Thanks, looks good!

File: Source/Core/sampleTerrain.js [hunk 2]
```diff
@@ -1,12 +1,12 @@
 /*global define*/
 define([
+        './Check',
         '../ThirdParty/when',
-        './defined',
-        './DeveloperError'
+        './defined'
```
Comment (pjcozzi) [inline Source/Core/sampleTerrain.js]: I don't think `defined` is needed anymore.

Comment (omh1280) [inline Source/Core/sampleTerrain.js]: @rms13 Patrick is right, could you remove those please?

Comment (rms13) [inline Source/Core/sampleTerrain.js]: Removed. Thanks!

**Pull request merged**